### DISCUSSION TITLE
Omit or change columns in list mode: is it possible?

### DISCUSSION THREAD

**Discussion by OSAwebsite** [code4recovery/12-step-meeting-list#1195] [Q&A] [answered]
For my particular use of the TSML list of meetings, I would like to be able to do one of two things:

1. Omit the column 'location/group'
or
2. Change the content of the title of the column and of the content in the column 'location/group'

My reasoning:
For me, the name of a group to a fault equals the name of the meeting.
As such, the 'group' in the column 'location/group' is redundant information.
The best solution for me would be the second and my change would be to omit the name of the group.
The location is useful, though.

Is this possible and is this easy to do?

Thanks,
Nicky

**Comment by OSAwebsite** [comment]
I have done an experiment and I found the following:
when I perform a search in the meeting list, using the open 'search' filter, a drop down appears with found matches.
I know realise that when a group name is shown in those matches, it comes from that (for me redundant) group name field.

I think that the thing I dislike most about the plugin is that nowhere is it documented where which input information is used implicitly.

If there is anything I can do to customise these things, that would be wonderful.

**Comment by joshreisner** [comment, marked as answer]
@OSAwebsite sure you can change it - here's the documentation: https://wordpress.org/plugins/12-step-meeting-list/#can%20i%20change%20the%20%22location%20%2F%20group%22%20column%20to%20display%20only%20the%20location%20name%20instead%3F

**Reply by joshreisner** [reply]
the example in the FAQ changes the default order of columns, here is a version that only changes the Location / Group column to Location

```php
$tsml_columns = [
  'time' => 'Time',
  'distance' => 'Distance',
  'name' => 'Meeting',
  'location' => 'Location',
  'address' => 'Address',
  'region' => 'Region',
  'district' => 'District',
  'types' => 'Types',
];
```

some of these columns only show in certain contexts:

* `distance` in a near location / near me search
* `district` in lieu of `region` when selecting "switch to districts" 
* `types` when enabled with CSS ([here is the recommended CSS for that from the FAQ](https://wordpress.org/plugins/12-step-meeting-list/#can%20i%20see%20types%20in%20the%20meeting%20list%3F%20and%20can%20i%20adjust%20the%20%2Fmen%20and%20%2Fwomen%20after%20the%20meeting%20name%3F))

any column should be able to be removed, but we don't recommend removing the `name` column because it's critical info, and needed in order to show links to the meeting detail pages

**Comment by OSAwebsite** [comment]
Just thinking out loud here: is there a reason why hard returns are not accepted in the group notes? The part in the bottom of the group details left column? That would be a huge help for me.

**Reply by joshreisner** [reply]
all the notes fields should allow returns within the text - [here is an example](https://code4recovery.org/meetings/meeting) - they are trimmed in the input sanitization process, so whitespace before or after is removed

**Comment by OSAwebsite** [comment]
Thanks - will have a look at that

**Comment by OSAwebsite** [comment]
What would I put in if I want to omit 'name'?

**Comment by OSAwebsite** [comment]
OK found that out. Just put in nothing.

**Comment by OSAwebsite** [comment]
It did not work for 'types'. The header is gone but the content is still there...

**Comment by OSAwebsite** [comment]
Is there a similar way to get rid of the District in the group section on the details page, at the bottom left?

**Reply by joshreisner** [reply]
yes you should be able to add some CSS to do that

```css
#tsml #meeting .group-district { display: none; }
```

**Comment by OSAwebsite** [comment]
Some of my experiences today... for what it is worth.

**Trash: difficult to empty**
 It is easy to trash a meeting, but emptying the trash is a nightmare. How long it takes is a wild guess. Something within 30 seconds, sometimes 5 minutes for 50 records, even when done one right after the other.

I also get errors. Today I tried out my changes on my staging site first. When I thought all was as expected, I confidently refreshed my entire list of meetings on the live website (while on maintenance). Only then did I see things I did not anticipate nor notice on the staging site. I had to reload the list of meetings 3 times because I kept seeing new things that were unexpected to me. So I had over 1000 meetings in de trash. There is a nice button titled 'Empty trash', so that is what I went for. I got an error. When trying to do a bulk delete of everything in the trash, the same error appeared and only a handful of meetings were deleted. Here is the error.

![Screenshot 2023-08-21 at 1 29 07 PM](https://github.com/code4recovery/12-step-meeting-list/assets/59859095/022ba95f-1ed8-4c88-9958-c71a23668581)

I ended up having to delete meetings per 50. I have had that experience before and wonder why this is happening.

**Imported information ignored**
The reason I wanted to empty the trash is this:
I had made changes in the group notes section. This information was present in my .csv, but whatever I did, it was ignored and the plugin kept coming up with the old information. This is when I started to suspect that some of the information in the trash may still be used. After I emptied the trash, the  imported information came through. Can you do some research to find out why this kept happening?

I did notice that, before I emptied the trash, the count of meetings and other content in the 'Where if my info" section kept including the trashed meetings so that might be a clue for you to find out why this is happening?

**Returns in notes fields**
You are right, Josh, they allow returns but only when they are formatted as line breaks. I forgot about that. Had that set up for the main notes fields, but not the group notes. My bad.

I think that's it. I hope it makes sense and that you can have a look at that behaviour of the trash?

thanks in advance,
Nicky

**Reply by joshreisner** [reply]
i'm not noticing any issues on our demo site emptying the trash. when i empty it with 39 meetings it takes less than a second. we do have some logic that runs when deleting a meeting - the plugin looks for and deletes any locations or groups that no longer have a meeting associated. but we are not running any permissions checks, so i suspect the error you saw was something else. maybe a plugin like wordfence was auditing it somehow?

things to know about groups: they are unique by name and are shared between meetings. so when editing a group that affects all meetings with that group. but this has an implication for importing: if you have several rows with the Group name `One Day at a Time` then it will only take the group notes / contacts / etc from ONE of those rows.

for what it's worth, this also applies to locations. these are unique by address. so all the meetings at `123 Main St` will have the same location notes. when importing, only the location notes / district / location name from ONE of the rows with that address will be used.

**Reply by joshreisner** [reply]
in case it's not clear, what i suspect was happening to you was you had multiple rows with a group but were only editing the group notes for one of the meetings, but the importer picked a different meeting, which is why it was getting lost

if the change is not shared by all meetings with that group, then that is a signal that group notes is the wrong field and youd be better off using the meeting notes field.

**Comment by OSAwebsite** [comment]
Suppressing the District number in the group notes with your suggested adjustment in custom CSS works in Chrome and Edge, but not in Safari. Safari seems often to hold on to stuff in cash much longer than the other two. We will see what it looks like tomorrow. This is a nice-to-have, not a must-have, by the way. Two out of three is not bad... :)

Thanks!

**Comment by OSAwebsite** [comment]
Thanks for that explanation. I have worked on the site all day to get it
right so I may be too tired to understand right now.
But, am I understanding correctly that the change in function.php is not
just affecting the headers, but also the content in that column?
Because when I took out the word 'group' from 'location/group', the group
disappeared from that column. And that is what I wanted.
However, I thought this was because I also left 'name' empty. The link to
the group was preserved, just the column header was gone.
Same with 'types'.
I have now put all back except 'group' because I really don't need that to
show up in that column as explained above.

Calling it a day now but if there is more to learn: teach me!

thanks :)


-
Yours in service,
Nicky D
Acting Ontario South Website Coordinator
al-anon.alateen.on.ca

*This too will pass...*

**Reply by joshreisner** [reply]
our default configuration uses

```php
  'location_group' => 'Location / Group',
```

this shows the location for in person meetings and the group for online meetings. when you switched it to 

```php
  'location' => 'Location',
```

it's the keyword `location` on the left side that changed the behavior, the right side is the column label. if you wanted to change this to 

```php
  'location' => 'Emplacement',
```

this should keep the same behavior but change the column header to say "Emplacement" instead. hope that clarifies it a little.
